services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/resource_tracker_core.py

```diff
@@ -0,0 +1,135 @@
+import datetime
+import json
+import logging
+from typing import Any
+
+import arrow
+from fastapi import FastAPI
+from models_library.users import UserID
+from prometheus_api_client import PrometheusConnect
+from simcore_service_resource_usage_tracker.modules.prometheus import (
+    get_prometheus_api_client,
+)
+
+# This script assumes everywhere that the minimum granularity of the data is 1 minute
+# Setting it smaller is unreasonable at least for prometheus, as the scraping interval is apprx. eq. to 1 h
+
+
+_logger = logging.getLogger(__name__)
+
+
+def _assure_dict_entry_exists(
+    metric_data, max_values_per_docker_id, image, userid
+) -> None:
+    for metric in metric_data:
+        current_id = metric["metric"]["id"]
+        if current_id not in max_values_per_docker_id.keys():
+            max_values_per_docker_id[current_id] = {
+                "container_uuid": metric["metric"]["container_label_uuid"],
+                "cpu_seconds": 0,
+                "uptime_minutes": 0,
+                "nano_cpu_limits": 0,
+                "egress_bytes": 0,
+                "image": image,
+                "user_id": userid,
+            }
+
+
+async def _evaluate_service_resource_usage(
```

Comment:
I see this function is async but all the calls to Prometheus are sync.
I would actually make the function sync and call it via loop.run_in_executor on the shared thread pool

Reply:
see comment above